### PR TITLE
bug fix: clear angle divisor for all geometries if instancing is used.

### PR DESCRIPTION
**Pull request by emnh**
This fixes the issue I mentioned in #5171 : when angle instancing is used together with regular geometries, there are strange problems, for example objects that do not render or textures are uniformly coloured by one of its pixel values.

I am not sure if this is the best way to fix it, but at least it works for now.

### CODE REVIEW

**Comment by benaadams** [comment]
Definately bug here; but the divisor reset should probably go in WebGLState, probably enableAttribute should take note of the divisor if set; and then reset back to 0 if not specified and changed > 0?

**Comment by emnh** [comment]
@benaadams I tried to do what you said in the 2nd commit.

Note: If I set state.usedInstancedBufferCount to 0 again after resetting then it did not work.

File: src/renderers/webgl/WebGLState.js [hunk 1]
```diff
@@ -78,6 +78,18 @@ THREE.WebGLState = function ( gl, extensions, paramThreeToGL ) {
 
 		}
 
+		if ( this.usedInstancedBufferCount > 0 ) {
+
+			var extension = extensions.get( 'ANGLE_instanced_arrays' );
+
+			if ( extension != null ) {
```
Comment (mrdoob) [inline src/renderers/webgl/WebGLState.js]: I suspect we don't need this check.

Comment (emnh) [inline src/renderers/webgl/WebGLState.js]: Ok. Fixed.

**Comment by benaadams** [comment]
Does ```extensions.get``` and ```extensions.vertexAttribDivisorANGLE``` for every attribute in the scene once instancing is used; and twice for the instanced attribute.

This does fix the issue but should be able to cut down the gpu comms a little more; having a look into it.

**Comment by benaadams** [comment]
This work better? #7230

**Comment by emnh** [comment]
@benaadams  Unfortunately your patch #7230 shows the same problem as without it..

**Comment by emnh** [comment]
Oh wait, I might have checked out the wrong branch.. just a sec

**Comment by emnh** [comment]
@benaadams It works now :)

**Comment by mrdoob** [comment]
#7230

**Pull request closed without merging**